geometric_kernels/spaces/hypersphere.py

```diff
@@ -170,3 +170,12 @@ def get_eigenvalues(self, num: int) -> B.Numeric:
             eigenfunctions.num_eigenfunctions_per_level,
         )  # [num,]
         return B.reshape(eigenvalues, -1, 1)  # [num, 1]
+
+    def ehess2rhess(self, x, egrad, ehess, direction):
+        """
```

Comment:
Maybe we should cite something? E.g. https://sites.uclouvain.be/absil/2013-01/Weingarten_07PA_techrep.pdf or some other source.